hugo/content/docs/smodel.md

```diff
@@ -0,0 +1,554 @@
+
+### alignFeature
+
+Controls the position adjustment of an element. It is given by the `alignment: Point` property of an element.
+It applies a `translate` to the svg element.
+
+*Interface:* `Alignable`
+
+```typescript
+export interface Alignable extends SModelExtension {
+    alignment: Point
+}
+```
+
+### boundsFeature
+
+Controls if the bounds of an element can be calculated and updated. The element needs a `position` and `size`.
+
+*Interface:* `BoundsAware`
+
+```typescript
+export interface BoundsAware extends SModelExtension{
+    bounds: Bounds
+}
+```
+
+### connectableFeature
+
+Controls if an element can be connected to other elements. To be connectable, the element must be an `SRoutableElementImpl` and have a role of source or target.
+
+*Interface:* `Connectable`
+
+```typescript
+
+export interface Connectable extends SModelExtension{
+    canConnect(routable: SRoutableElementImpl, role: 'source' | 'target'): boolean
+}
+```
+
+### creatingOnDragFeature
+
+Controls if an element is created on drag. The element needs to implement a `createAction` method that returns an `Action` that creates the element.
+
+*Interface:* `CreatingOnDrag`
+
+```typescript
+export interface CreatingOnDrag extends SModelExtension {
+    createAction(id: string): Action
+}
+```
+
+### decorationFeature
+
+Controls if an element is a decoration.
+
+*Interface*: `Decoration`
+
+```typescript
+export interface Decoration extends SModelExtension {}
+```
+
+### deletableFeature
+
+Controls if an element can be deleted from the diagram. To be deletable, an element needs to be an `SChildElementImpl`.
+
+*Interface:* `Deletable`
+
+```typescript
+export interface Deletable extends SModelExtension {}
+```
+
+### edgeLayoutFeature
+
+Controls if an element is layoutable on an edge. The element must be an `SChildElementImpl`, its parent must be an `SRoutableElementImpl`, the element must have the `edgePlacement` property, and must have the `boundsFeature`. This is commonly used for labels that are displayed on an edge.
+
+```typescript
+edgePlacement: {
+    position: number, // between 0 and 1, from the source (0) to the target (1)
+    side: 'bottom' | 'top' | 'left' | 'right' | 'on',
+    rotate: boolean
+}
+```
+
+*Interface:* `EdgeLayoutable`
+
+```typescript
+export interface EdgeLayoutable extends SModelExtension {
+    edgePlacement: EdgePlacement
+}
+```
+
+### editFeature
+
+Controls if an edge routing can be edited. The edge must be an `SRoutableElementImpl`.
+
+### editLabelFeature
+
+Controls if a label can be edited. The label must have the `text` property.
+
+*Interface:* `EditableLabel`
+
+```typescript
+export interface EditableLabel extends SModelExtension {
+    text: string
+    readonly isMultiline?: boolean
+    readonly editControlDimension?: Dimension
+    readonly editControlPositionCorrection?: Point
+}
+```
+
+### expandFeature
+
+Controls if an element can be expanded. The element must have the property `expanded`.
```

Comment:
manually expanded by the user? what is the purpose? hide the children of arbitrary SModelElements?